testslide/lib.py

```diff
@@ -20,26 +17,3 @@ def _bail_if_private(candidate: str, allow_private: False):
             "Please consider using patterns like dependency injection instead. "
             "If you really need to do this use the allow_private=True argument."
         )
-
-
-def _validate_function_signature(argspec: inspect.FullArgSpec, args, kwargs):
```

Comment:
IIRC, the reason for this to live here, was because we should also in the future, use it for `mock_callable()` as well.

Reply:
I had to move it because I'm using the class `StrictMock` to match the argument against it, so that would have introduced a cyclic import.

I could try to pass to the function the list of mock classes to check or similar, but seemed simpler to just move than create more layers.

Reply:
@fornellas yes, that was the exact reason.

Reply:
I'll try to add the "mock_extractors" param or similar with the list of mock classes and how to get the spec/template out of them then, though to be fair, `mock_callable` already depends on `strict_mock`.